### PR TITLE
Update Person.java and respective test cases

### PR DESCRIPTION
**Pull request by TanWenCong**
Person has no BirthdayMonth field.

This is not the best representation of our customers. Customers have a birthday month and we can give them discounts during their birthday months.

Let's include BirthdayMonth field into the Person class, so that we can give birthday discounts to customers accordingly.

Additional information for this commit:
Updated testcases for PersonTest, AddCommandParserTest, LogicManagerTest and other utility tests.

### CODE REVIEW

File: src/test/data/JsonAddressBookStorageTest/invalidAndValidPersonAddressBook.json [hunk 1]
```diff
@@ -3,11 +3,13 @@
     "name": "Valid Person",
     "phone": "9482424",
     "email": "hans@example.com",
-    "address": "4th street"
+    "birthdayMonth": "1",
+    "reward": "1234"
```
Comment (bensohh) [inline src/test/data/JsonAddressBookStorageTest/invalidAndValidPersonAddressBook.json:7]: I like how you updated the sample data according to the new implementation 👍

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -66,10 +67,25 @@ public static Phone parsePhone(String phone) throws ParseException {
...
-     * @throws ParseException if the given {@code address} is invalid.
+     * @throws ParseException if the given {@code birthdayMonth} is invalid.
+     */
+    public static BirthdayMonth parseBirthdayMonth(String birthdayMonth) throws ParseException {
+        requireNonNull(birthdayMonth);
+        String trimmedBirthdayMonth = birthdayMonth.trim();
+        if (!BirthdayMonth.isValidBirthdayMonth(trimmedBirthdayMonth)) {
+            throw new ParseException(BirthdayMonth.MESSAGE_CONSTRAINTS);
+        }
+        return new BirthdayMonth(trimmedBirthdayMonth);
```
Comment (bensohh) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:81]: I think the current implementation is very good. Perhaps if time permits, could consider converting integers 1 - 12 to the respective months Jan-Dec to be displayed as an enhancement 😄

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java [hunk 3]
```diff
@@ -41,59 +44,83 @@ public void toModelType_validPersonDetails_returnsPerson() throws Exception {
...
+    }
+
+    @Test
+    public void toModelType_nullBirthdayMonth_throwsIllegalValueException() {
+        JsonAdaptedPerson person = new JsonAdaptedPerson(VALID_NAME, VALID_PHONE, VALID_EMAIL,
+            null, VALID_REWARD, VALID_TAGS);
+        String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, BirthdayMonth.class.getSimpleName());
+        assertThrows(IllegalValueException.class, expectedMessage, person::toModelType);
+    }
+
```
Comment (bensohh) [inline src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java:110]: It is very commendable that you already updated the test cases for the implementation made. Keep up the good work!

**Review by bensohh — APPROVED** [review]
LGTM! Since this PR is quite big perhaps we have to do more extensive manual/automated testing to ensure that it is break-proof 👍

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 2 above)
Comment (TanWenCong) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:81]: Was thinking about this as well, but am not sure whether it'll be too packed to display birthday month in the UI together with the rest of the details. Will discuss with the team!

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java (hunk 3 above)
Comment (TanWenCong) [reply src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java:110]: Thanks a lot! :^)

File: src/test/data/JsonAddressBookStorageTest/invalidAndValidPersonAddressBook.json (hunk 1 above)
Comment (TanWenCong) [reply src/test/data/JsonAddressBookStorageTest/invalidAndValidPersonAddressBook.json:7]: Thanks :^)

**Pull request merged**